.gitignore

```diff
@@ -158,3 +158,6 @@ cython_debug/
 
 # vim
 *.swp
+
+#pre-commit
+.pre-commit-config.yaml
```

Comment:
Note that you can also add gitignores to your local global gitignore (not that I haven't been guilty of tossing my own local things into our project gitignores).

If you're using a a pre-commit hook to blacken code (or to use some other code formatter), this can be annoying since we don't actually require a black code style. It means that PRs from you will reformat other people's code, meaning that reviewing your PRs will involve looking over changed lines that are just style changes. If you can configure your reformatter to only reformat lines you've edited, that would make things easier.

(I see a few changes in here that look like black or YAPF decided to change things that didn't match its style preferences.)